### ISSUE TITLE
Looks like a Graph endpoint error

### ISSUE THREAD

**Issue by Arne-RFA** [Micke-K/IntuneManagement#148]
Deployment of our "templated" policy to a new tenant errored as per the below, I have obfuscated the request IDs

```
Failed to invoke MS Graph with URL https://graph.microsoft.com/beta/groups (Request ID: <GUID for request ID>). Status code: BadRequest. Response message: Cannot Create a mail-enabled security groups and or distribution list. Exception: The remote server returned an error: (400) Bad Request.
Create AAD Group For Placement Only - do not use for production (devices)
WARNING: Export folder for dependency ScopeTags not found
WARNING: Export folder for dependency Applications not found
WARNING: Export folder for dependency TermsOfUse not found
Import Conditional Access object Std Policy - All Apps - 2FA
Failed to invoke MS Graph with URL https://graph.microsoft.com/beta/identity/conditionalAccess/policies (Request ID: <different to above GUID>). Status code: BadRequest. Response message: The server could not process the request because it is malformed or incorrect. Exception: The remote server returned an error: (400) Bad Request.
```

Reproduced the error on a clean Dev tenant.

**Comment by Micke-K** [comment]
Hello!

Looks like it failed to create a Azure AD group. Not sure why and if that caused the error when creating the CA. You could try to create the group manually and see if that fixed it. 

Do you have any additional settings in the CA? Looks like a MF CA. Do you use 3rd party CA or custom CA policy? Do you have anything else like Terms of Use, Filter etc?

Cheers?

**Comment by Arne-RFA** [comment]
Hello!

Apologies, forgot to confirm that this is on 3.9

The std deployment includes ~20 CAP (Conditional Access Policies, to distinguish from CA which will always be a "Certificate Authority" in my head) many of which use the same groups.

When the issue was first seen, the issue was internally reported to me as a single CAP out of the ~20 that had not deployed as part of a bulk import, with the same issue on two newly created CSP deployed tenants.

When I reproduced the issue, I reproduced in a clean (newly created) Dev tenant (the free one with 25 M365 E5 licences) importing just this policy (which would also create the groups).

Although only the single CAP was selected to import into the Dev tenant, all of the expected (1) groups in the export were created, but the single selected CAP was not created.

(1) checking the export, there is a file "Compliance Notifications.json" in the Groups directory, but this is not a visible group in the "template" tenant. The "Compliance Notifications.json" file does contain "mail" and "proxyAddresses" attributes with addresses that are only in the template tenant, but I think this might be a "red herring"

.json of the CAP that failed to deploy as below

```
{
    "@odata.context":  "https://graph.microsoft.com/beta/$metadata#identity/conditionalAccess/policies/$entity",
    "@odata.type":  "#microsoft.graph.conditionalAccessPolicy",
    "@odata.id":  "identity/conditionalAccess/policies(\u00270656c5cc-5bac-4cc4-aa13-e4cc1e9a127c\u0027)",
    "@odata.editLink":  "identity/conditionalAccess/policies(\u00270656c5cc-5bac-4cc4-aa13-e4cc1e9a127c\u0027)",
    "id":  "0656c5cc-5bac-4cc4-aa13-e4cc1e9a127c",
    "templateId":  null,
    "displayName":  "Std Policy - All Apps - 2FA",
    "createdDateTime@odata.type":  "#DateTimeOffset",
    "createdDateTime":  "2021-07-23T08:05:02.0135927Z",
    "modifiedDateTime@odata.type":  "#DateTimeOffset",
    "modifiedDateTime":  "2023-06-22T15:30:27.4714346Z",
    "state@odata.type":  "#microsoft.graph.conditionalAccessPolicyState",
    "state":  "enabled",
    "sessionControls":  null,
    "conditions":  {
                       "@odata.type":  "#microsoft.graph.conditionalAccessConditionSet",
                       "userRiskLevels@odata.type":  "#Collection(microsoft.graph.riskLevel)",
                       "userRiskLevels":  [

                                          ],
                       "signInRiskLevels@odata.type":  "#Collection(microsoft.graph.riskLevel)",
                       "signInRiskLevels":  [

                                            ],
                       "clientAppTypes@odata.type":  "#Collection(microsoft.graph.conditionalAccessClientApp)",
                       "clientAppTypes":  [
                                              "all"
                                          ],
                       "platforms":  null,
                       "times":  null,
                       "deviceStates":  null,
                       "devices":  null,
                       "clientApplications":  null,
                       "applications":  {
                                            "@odata.type":  "#microsoft.graph.conditionalAccessApplications",
                                            "includeApplications@odata.type":  "#Collection(String)",
                                            "includeApplications":  [
                                                                        "All"
                                                                    ],
                                            "excludeApplications@odata.type":  "#Collection(String)",
                                            "excludeApplications":  [
                                                                        "0000000a-0000-0000-c000-000000000000",
                                                                        "d4ebce55-015a-49b5-a083-c84d1797ae8c"
                                                                    ],
                                            "includeUserActions@odata.type":  "#Collection(String)",
                                            "includeUserActions":  [

                                                                   ],
                                            "includeAuthenticationContextClassReferences@odata.type":  "#Collection(String)",
                                            "includeAuthenticationContextClassReferences":  [

                                                                                            ],
                                            "applicationFilter":  null
                                        },
                       "users":  {
                                     "@odata.type":  "#microsoft.graph.conditionalAccessUsers",
                                     "includeUsers@odata.type":  "#Collection(String)",
                                     "includeUsers":  [
                                                          "All"
                                                      ],
                                     "excludeUsers@odata.type":  "#Collection(String)",
                                     "excludeUsers":  [

                                                      ],
                                     "includeGroups@odata.type":  "#Collection(String)",
                                     "includeGroups":  [

                                                       ],
                                     "excludeGroups@odata.type":  "#Collection(String)",
                                     "excludeGroups":  [
                                                           "c38a21c7-ad91-4f09-b158-594d7b3c7fad",
                                                           "ab5e5204-380a-4da6-bb36-ac1835a23082"
                                                       ],
                                     "includeRoles@odata.type":  "#Collection(String)",
                                     "includeRoles":  [

                                                      ],
                                     "excludeRoles@odata.type":  "#Collection(String)",
                                     "excludeRoles":  [

                                                      ],
                                     "includeGuestsOrExternalUsers":  null,
                                     "excludeGuestsOrExternalUsers":  null
                                 },
                       "locations":  {
                                         "@odata.type":  "#microsoft.graph.conditionalAccessLocations",
                                         "includeLocations@odata.type":  "#Collection(String)",
                                         "includeLocations":  [
                                                                  "All"
                                                              ],
                                         "excludeLocations@odata.type":  "#Collection(String)",
                                         "excludeLocations":  [
                                                                  "05f3db25-01bc-4ec5-8d27-c94ee619feb6"
                                                              ]
                                     }
                   },
    "grantControls":  {
                          "@odata.type":  "#microsoft.graph.conditionalAccessGrantControls",
                          "operator":  "OR",
                          "builtInControls@odata.type":  "#Collection(microsoft.graph.conditionalAccessGrantControl)",
                          "builtInControls":  [
                                                  "mfa"
                                              ],
                          "customAuthenticationFactors@odata.type":  "#Collection(String)",
                          "customAuthenticationFactors":  [

                                                          ],
                          "termsOfUse@odata.type":  "#Collection(String)",
                          "termsOfUse":  [

                                         ],
                          "authenticationStrength@odata.context":  "https://graph.microsoft.com/beta/$metadata#identity/conditionalAccess/policies(\u00270656c5cc-5bac-4cc4-aa13-e4cc1e9a127c\u0027)/grantControls/authenticationStrength/$entity",
                          "authenticationStrength@odata.associationLink":  "https://graph.microsoft.com/beta/identity/conditionalAccess/policies(\u00270656c5cc-5bac-4cc4-aa13-e4cc1e9a127c\u0027)/grantControls/authenticationStrength/$ref",
                          "authenticationStrength@odata.navigationLink":  "https://graph.microsoft.com/beta/identity/conditionalAccess/policies(\u00270656c5cc-5bac-4cc4-aa13-e4cc1e9a127c\u0027)/grantControls/authenticationStrength",
                          "authenticationStrength":  null
                      }
}

```

**Comment by Micke-K** [comment]
Hello!

The CAP has two apps excluded, two groups excluded and one location excluded. 

I would assume one of them causes the failure. 

About the import of all groups...yes, it is annoying. I don't like that it imports all groups. I started looking at it a while back but I have to re-write the whole logic for group import to fix it. It's on my todo list.

Cheers!

**Comment by Micke-K** [comment]
You could try to remove the IDs for the apps and location from the json before import. 

Be careful with the import so you don't lock yourself out when removing exclusions. 

Cheers!

**Comment by Arne-RFA** [comment]
a better look at the log shows that the failed group is the Compliance Notification group, so is a red herring

I removed the exclude groups from the .json and attempted import again (with CAP as off), and still get the error.
I then ran a new export from the template tenant, retried using the new export and got the same error, removed the exclude groups and still got the same error.

Looking at it the other way round, I went to create the CAP manually, and think I might have found the issue.
The two excluded apps are Microsoft Intune (0000000a-0000-0000-c000-000000000000) and Microsoft Intune Enrollment (d4ebce55-015a-49b5-a083-c84d1797ae8c) so that if we import into an existing tenant, both are covered, however in a new tenant, only Microsoft Intune is enabled.

I tried removing the Intune Enrollment GUID from the json and it still failed (passes as valid json on a linter), removed the  Intune Enrollment App in the template tenant, and ran another export, then imported, and it worked.

Comparing the two .json, apart from the date, it looks as if jsonlint.com is more relaxed than Microsoft for what comprises valid json.

So it appears that an app that is excluded (or presumably included) in a CAP must already exist in the tenant.

When I get a little more time, I will test adding the Intune Enrollment app again to the CAP and then trying import and export again.
